File: in-motion-ios/inMotion/inMotion/Models/DTO/Post/Incoming/GetPostResponseDto.swift
//
//  GetPostResponseDto.swift
//  inMotion
//
//  Created by Kamil Pietrak on 31/10/2023.
//

import Foundation

struct GetPostResponseDto: Codable {
    public var id: UUID
    
    public var description: String
    public var title: String
    
    public var author: PostAuthorDto
    public var tags: [PostTagDto]
    
    public var videos: [PostVideoDto]
    public var postCommentsCount: UInt
    public var postReactionsCount: UInt
    
    public var isLikedByUser: Bool
    public var postReaction: PostReactionWithoutAuthorDto?
    
    public var createdAt: Date
}

File: in-motion-android/app/src/main/java/com/inmotion/in_motion_android/data/remote/dto/posts/GetPostResponseDto.kt
package com.inmotion.in_motion_android.data.remote.dto.posts

import java.io.Serializable

data class GetPostResponseDto(
    val id: String,
    val description: String?,
    val title: String?,
    val author: PostAuthorDto,
    val tags: List<PostTagDto>,
    val videos: List<PostVideoDto>,
    val isLikedByUser: Boolean,
    val postReaction: PostReactionWithoutAuthorDto,
    val postCommentCount: Int,
    val postReactionsCount: Int,
    val createdAt: String
): Serializable
